compiler/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -833,6 +852,35 @@ impl<'interner> TypeChecker<'interner> {
                     }
                 }
             }
+            Type::NamedGeneric(_, _) => {
+                let func_meta = self.interner.function_meta(&self.current_function.unwrap());
+
+                for constraint in func_meta.trait_constraints {
+                    if let Some(trait_id) = constraint.trait_id {
```

Comment:
I had it as an Option since a trait could potentially fail to resolve, but you're right this doesn't make sense in the HIR struct. Fix: https://github.com/noir-lang/noir/pull/2717/commits/3e59ecad8d9b1dac7035d6fd067fb18f02f8a6b1